README.md

```diff
@@ -1,3 +1,30 @@
 # Workleap.OpenApi.MSBuild
 
 Validates at build time that the OpenAPI specification files extracted from the ASP.NET Core Web API being built conform to Workleap API guidelines.
+
+Depending if the users the Contract-First or Code-First approach this MsBuild task will:
+
+- Install tools: OasDiff, Spectral, SwashbuckleCLI
+- Generate the OpenAPI specification file from the associated Web API
+- Validate Spectral rules
+- Compare the given OpenAPI specification file with the generated one
+
+## How it work
+
+- The entry point is `ValidateOpenApiTask.ExecuteAsync()` and will be executed during the build process on project that reference it. This is defined in `./src/Workleap.OpenApi.MSBuild/msbuild/tools/Workleap.OpenApi.MSBuild.targets` as a `UsingTask.TaskName`
```

Comment:
Maybe link to the official MSBuild docs? https://learn.microsoft.com/en-us/visualstudio/msbuild/tutorial-custom-task-code-generation?view=vs-2022#include-msbuild-properties-and-targets-in-a-package